### PR TITLE
Firmware Update - Simple Update

### PR DESCRIPTION
**Pull request by MaciejMis**
Hi!
This is an implementation of Redfish Simple Update. Device Manager will send a Simple Update request to a given device.

**Note:** This is a generic implementation of Redfish Simple Update. Specific vendors might require additional parameters.

Firmware Update White Paper: https://www.dmtf.org/sites/default/files/standards/documents/DSP2062_1.0.0.pdf

|Parameter|Type|Description|Required|
|-------|--------------------|-------------------|-------|
|`ImageURI`|String|The URI of the image to download and install.|**true**|
|`TransferProtocol`|String|The network protocol that the update service uses to retrieve the image file located at the URI provided in the ImageURI parameter. <br>**Note**: This parameter is only used when the scheme is not encoded in the ImageURI parameter. If the scheme is present, this parameter is ignored.|**false**|
|`Targets`|Array|An array of strings that are URIs to resources that indicate where to apply the image.|**false**|
|`Username`|String|The user name to access the URI specified by the `ImageURI` parameter.|**false**|
|`Password`|String|The password to access the URI specified by the `ImageURI` parameter.|**false**|

### CODE REVIEW

File: src/update_service.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	response, body, _, postErr := postHTTPDataByRfAPI(ipAddress, SimpleUpdateURI, userData, request)
+	if postErr != nil {
+		logrus.Errorf("error during http post to redfish device: %s", postErr.Error())
+		return "", err
+	}
+
+	// Redfish SimpleUpdate should return 202 Accepted status code. In addition, Task Monitor should be present in Location header.
+	statusCode := response.StatusCode
```
Comment (ehaligow) [inline src/update_service.go:47]: Maybe we can get status code from postHTTPDataByRfAPI? Instead from response.

File: src/update_service.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		logrus.Errorf("json.Marshal returned error: %s", err.Error())
+		return ""
+	}
+
+	return string(request)
+}
+
+func validateSimpleUpdateRequest(request SimpleUpdateRequest) error {
+	if len(request.ImageURI) == 0 {
+		errorMsg := "image URI is required for SimpleUpdate"
```
Comment (ehaligow) [inline src/update_service.go:73]: Is it necessary to introduce new variable?

Comment (MaciejMis) [reply src/update_service.go]: No, fixed.

**Comment by jcleung5549** [comment]
To make testing this pull request easier, the executable should be placed in a folder (e.g. "exe", "bin") so each person who wants to test doesn't need to perform the build. A simple set of instructions would be good.

**Comment by MaciejMis** [comment]
This PR was approved at the OPENRMC-DM meeting.

**Pull request merged**